server/data/data-jmap/src/test/java/org/apache/james/jmap/api/change/EmailChangeRepositoryContract.java

```diff
@@ -98,6 +98,39 @@ default void getLatestStateShouldReturnLastPersistedState() {
             .isEqualTo(change3.getState());
     }
 
+    @Test
+    default void getLatestStateShouldReturnLastNonDelegatedPersistedState() {
+        EmailChangeRepository repository = emailChangeRepository();
+
```

Comment:
How about a nested class for all the delegation stuff? Would be less confusing since most of the tests are the same as non-delegated ones.

Reply:
Was my first thought but `emailChangeRepository()` can not be refered in a static context hence  gave up (class in interfaces have an implicit static qualifier)